core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -469,20 +475,33 @@ private void handleCacheStreamException(IOException e) {
 
   /**
    * Only updates {@link #mCurrentCacheStream}, {@link #mCurrentBlockInStream} and
-   * {@link #mStreamBlockId} to be in sync with the current block (i.e. {@link #getCurrentBlockId()}
-   * ). If this method is called multiple times, the subsequent invokes become no-op. Call this
-   * function every read and seek unless you are sure about the block streams are up-to-date.
+   * {@link #mBlockIdIndex} to be in sync with the current block
+   * (i.e. {@link #getCurrentBlockId()}). If this method is called multiple times, the subsequent
+   * invokes become no-op. Call this function every read and seek unless you are sure about the
+   * block streams are up-to-date.
    */
   private void updateStreams() throws IOException {
+    if (shouldUpdateStreams(mBlockIdIndex)) {
+      mBlockIdIndex++;
```

Comment:
is it possible that the position is already at the end of the file, so incrementing the index would overflow?

Reply:
yes, but `getCurrentBlockId` will return `-1` in that case, so `updateStreamsBasedOnBlockIdIndex` will handle the case correctly.

Reply:
I improved the logic, now, the mBlockIdIndex will not overflow.